python/example/network_ring_mpi.py

```diff
@@ -31,15 +31,15 @@ def make_cable_cell(gid):
     b0 = tree.append(s, arbor.mpoint(0, 0, 0, 2), arbor.mpoint(50, 0, 0, 2), tag=3)
 
     # Attach two dendrites (tag=3) of length 50 μm to the end of the first dendrite.
-    # Radius tapers from 2 to 0.5 μm over the length of the dendrite.
-    b1 = tree.append(
+    # (b1) Radius tapers from 2 to 0.5 μm over the length of the dendrite.
```

Comment:
No, that's not good. In an example, unused variables are OK, they are they to facilitate quick experimentation, not lessons of writing optimized code. Users way want to inspect the return value of `.append()`.

Reply:
Well, this is not about performance. Flake8 complains as unused variables often indicate an oversight.
Example:
```py
b0 = tree.append(None, ...)
b1 = tree.append(b0, ...)
b2 = tree.append(b0, ...)
```
`b1` goes unused, so is this a typo or CopyPaste bug, and the user really meant
```py
b0 = tree.append(None, ...)
b1 = tree.append(b0, ...)
b2 = tree.append(b1, ...)
```

Reply:
> No, that's not good. In an example, unused variables are OK, they are they to facilitate quick experimentation, 

If that is inteded, we have the options of
1. Setting a file-wide igore for that warning
2. Suppressing it in this lines via `#noqu:XXX`

I'd just roll with it, since if experimentation is desired, adding a `x =` to the line shouldn't be to hard. We already
demonstrated that is possible above.

Reply:
> Flake8 complains as unused variables often indicate an oversight.

In 'production' code, sure, but this isn't. (It isn't a copypaste bug either.)

Since it's my goto tutorial, I know that users only vaguely familiar with Python (i.e. scientific users ;)) can and are surprised by anything that's not systematic. 

Let's clarify why we don't assign then:

```suggestion
    # (b1) Radius tapers from 2 to 0.5 μm over the length of the dendrite. Because we're not going to reference this segment later on, we don't need to assign it to a variable.
```